src/main/cpp/src/zorder.cu

```diff
@@ -27,6 +27,118 @@
 #include <thrust/iterator/constant_iterator.h>
 #include <thrust/iterator/counting_iterator.h>
 
+namespace {
+
+// pretends to be an array of uint32_t, but really only stores
+// the data in a long with a set number of bits allocated for
+// each item (num_bits_per_entry)
+struct long_backed_array {
+  long_backed_array() = delete;
+  ~long_backed_array() = default;
+  long_backed_array(long_backed_array const&) = default;  ///< Copy constructor
+  long_backed_array(long_backed_array&&) = default;  ///< Move constructor
+  inline __device__ explicit long_backed_array(int32_t num_bits_per_entry): data(0),
```

Comment:
Also we can avoid redundant `inline` keyword. When a function only has definition (implemented without a separate declaration) then it is implicitly inline. So all the `inline` keyword in this file can be removed.